webserver.py

```diff
@@ -362,6 +368,31 @@ def harvest():
     return jsonify(game_state)
 
 
+@app.route('/action/unlock', methods = ['GET', 'POST'])
+def unlock():
+    # requesting unlock action data from script (x, y)
+    data = request.json
+
+    # loading game_state
+    game_state = load_state(data['slug'])
+
+    # safety check
+    if data['password'] != game_state['password']:
+        raise Exception("Invalid password")
+    if game_state['cash'] < GAME_CONFIG['plotPrice'] * GAME_CONFIG['plotMultiplier'] ** game_state['unlockCount']:
+        raise Exception("Not enough cash.")
```

Comment:
I believe this will show up as a server error rather than a message to the user.  Instead, you want to send them a 403 with the message.